### DISCUSSION TITLE
esp32-pico died.

### DISCUSSION THREAD

**Discussion by mcfoster** [thehighnibble/thehighnibble.github.io#139] [IMSAI8080esp replica]
This is a great kit, but I think I killed my esp32 with a firmware update.
I tried reflashing it but I get the following error:
```
$ esptool --chip esp32 --port /dev/ttyUSB0 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xd000 ota_data_initial.bin 0x1000 bootloader.bin 0x10000 imsaisim_esp32.bin 0x8000 part_table.bin
esptool.py v2.8
Serial port /dev/ttyUSB0
Connecting........_____.....__
Chip is ESP32-PICO-D4 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, Embedded Flash, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: d8:a0:1d:5a:12:a4
Changing baud rate to 921600
Changed.
Enabling default SPI flash mode...
Configuring flash size...
Warning: Could not auto-detect Flash size (FlashID=0xffffff, SizeID=0xff), defaulting to 4MB
Erasing flash...
Compressed 8192 bytes to 31...

A fatal error occurred: Timed out waiting for packet header

```
Any ideas?

**Comment by MKoenen1** [comment]
It looks like you are using an old version of ESPTOOL.PY V2.8  I just updated my new ESP32-PICO-D4 to V1.12 and I used the ESPTOOL.PY version 4.7 which I downloaded using PIP.  You should try the newer version.  I did not get the warning message about could not detect flash size during my update.  My update said "Auto-detected Flash size: 4MB".

I hope this helps.

**Reply by mcfoster** [reply]
Thank you. That did the trick for the esp32 flashing. 
I still have another issue related to the sd card I think, but you got me over that problem.

**Reply by MKoenen1** [reply]
Thanks, Just checking, did you solder a connector to the 6 pins on the ESP32?  Seems something that could be missed.

**Reply by mcfoster** [reply]
Thanks for the suggestions.  After updating my ESPTOOL.PY application as you recommended, and re-imaging the SD card I am back up and running.   I will be extra careful next time I decide to upgrade.
